packages/components/src/DataList/DataList.utils.tsx

```diff
@@ -102,3 +102,23 @@ export function sortSizeProp(sizeProp: Breakpoints[]) {
     (a, b) => BREAKPOINTS.indexOf(a) - BREAKPOINTS.indexOf(b),
   );
 }
+
+export function getExposedActions(
+  childrenArray: ReactElement[],
+  childCount = 2,
+) {
+  const firstNChildren = childrenArray.slice(0, childCount);
+
+  return firstNChildren.reduce((result: typeof childrenArray, child, i) => {
+    const hasIcon = Boolean(child.props.icon);
+
+    if (!hasIcon) return result; // If the child does not have an icon, continue.
+
+    // If it's the first child or if the previous child was added, then add this child.
+    if (i === 0 || (i < childCount && result.length === i)) {
```

Comment:
that `result.length === i` confused me a bit there haha. mind putting that in a const and name it in a descriptive way